cap/modules/deposit/api.py

```diff
@@ -115,3 +116,20 @@ def clear(self, *args, **kwargs):
     #             raise  # TODO raise REST exception ...
 
     #     return super(CAPDeposit, self).publish(*args, **kwargs)
+
+    @index
+    @mark_as_action
+    def clone(cls, pid=None, id_=None):
+        """Clone a deposit.
+
+        Adds snapshot of the files when deposit is cloned.
+        """
+        data = copy.deepcopy(cls.dumps())
+        del data['_deposit']
+        deposit = super(CAPDeposit, cls).create(data, id_=id_)
+        deposit['_deposit']['cloned_from'] = pid.pid_value
```

Comment:
We should do something like in https://github.com/inveniosoftware/invenio-deposit/blob/master/invenio_deposit/api.py#L274-L278

```python
deposit['_deposit']['cloned_from'] = {
    'type': pid.pid_type,
    'value': pid.pid_value,
    'revision_id': self.revision_id,
}
```